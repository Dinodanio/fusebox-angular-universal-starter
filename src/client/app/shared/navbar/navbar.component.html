<nav class="mat-elevation-z6">
  <a class="nav-button" md-button routerLink>FUSEBOX-ANGULAR-UNIVERSAL-STARTER</a>
  <div class="flex-spacer"></div>


  <button md-button [mdMenuTriggerFor]="pages">Pages</button>

  <md-menu #pages="mdMenu">
    <button md-menu-item [mdMenuTriggerFor]="standard">Standard</button>
    <button md-menu-item [mdMenuTriggerFor]="seo">SEO</button>
  </md-menu>

  <md-menu #standard="mdMenu">
    <a mat-menu-item *ngFor="let item of navbarService.menu$ | async" [routerLink]="item.route" routerLinkActive="active" [routerLinkActiveOptions]="{exact: true}">
      {{ item.name }}
    </a>
  </md-menu>

  <md-menu #seo="mdMenu">
    <a mat-menu-item href="/sitemap.xml">sitemap.xml</a>
    <a mat-menu-item href="/robots.txt">robots.txt</a>
  </md-menu>

  <a class="nav-button" md-button href="/api-docs">API</a>
</nav>