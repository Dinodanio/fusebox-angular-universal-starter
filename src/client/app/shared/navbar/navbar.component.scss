:host {
  nav {
    background-color: #2196F3;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 8px 16px;
    color: rgba(255, 255, 255, 0.87);
  }
  img {
    width: 96px;
  }
  .flex-spacer {
    -webkit-box-flex: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
  .nav-button {
    //text-transform: uppercase;
  }
}