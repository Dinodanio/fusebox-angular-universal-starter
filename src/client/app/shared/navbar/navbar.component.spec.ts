import { RouterTestingModule } from '@angular/router/testing'
import { NavbarComponent } from './navbar.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { INavbarService, NavbarService } from './navbar.service'
import { Component } from '@angular/core'
import { MdButtonModule, MdMenuModule } from '@angular/material'
import '../../../operators'

describe(NavbarComponent.name, () => {
  let fixture: ComponentFixture<NavbarComponent>
  let navbarService: INavbarService

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, MdButtonModule, MdMenuModule],
      declarations: [NavbarComponent, TestComponent],
      providers: [NavbarService]
    }).compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(NavbarComponent)
    navbarService = TestBed.get(NavbarService)
  })

  afterEach(() => {
    TestBed.resetTestingModule()
  })

  test.skip('should compile', async(() => {
    fixture.detectChanges()
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture.nativeElement).toMatchSnapshot()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-navbar></pm-navbar>'
})
class TestComponent { }
