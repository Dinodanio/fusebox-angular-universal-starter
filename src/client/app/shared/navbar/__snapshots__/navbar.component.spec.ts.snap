// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarComponent should compile 1`] = `
<div
  id="root0"
  ng-version="5.0.0-rc.0"
>
  <nav
    class="mat-elevation-z6"
  >
    
  
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/"
      md-button=""
      ng-reflect-router-link=""
      routerlink=""
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        FUSEBOX-ANGULAR-UNIVERSAL-STARTER
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        mat-ripple=""
        ng-reflect-_mat-ripple-centered="false"
        ng-reflect-_mat-ripple-disabled="false"
        ng-reflect-_mat-ripple-trigger="[object HTMLAnchorElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    
  
    <div
      class="flex-spacer"
    />
    


  
    <button
      aria-haspopup="true"
      class="mat-button"
      md-button=""
      ng-reflect-menu="[object Object]"
    >
      <span
        class="mat-button-wrapper"
      >
        Pages
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        mat-ripple=""
        ng-reflect-_mat-ripple-centered="false"
        ng-reflect-_mat-ripple-disabled="false"
        ng-reflect-_mat-ripple-trigger="[object HTMLButtonElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    

  
    <md-menu>
      
    </md-menu>
    

  
    <md-menu>
      
    </md-menu>
    

  
    <md-menu>
      
    </md-menu>
    

  
    <a
      aria-disabled="false"
      class="nav-button mat-button"
      href="/api-docs"
      md-button=""
      tabindex="0"
    >
      <span
        class="mat-button-wrapper"
      >
        API
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        mat-ripple=""
        ng-reflect-_mat-ripple-centered="false"
        ng-reflect-_mat-ripple-disabled="false"
        ng-reflect-_mat-ripple-trigger="[object HTMLAnchorElement]"
      />
      <div
        class="mat-button-focus-overlay"
      />
    </a>
    

  </nav>
</div>
`;
